pkg/apis/kappctrl/v1alpha1/types_template.go

```diff
@@ -99,3 +100,9 @@ type AppTemplateSopsAge struct {
 type AppTemplateSopsPrivateKeysSecretRef struct {
 	Name string `json:"name,omitempty" protobuf:"bytes,1,opt,name=name"`
 }
+
+// +k8s:openapi-gen=true
+type AppTemplateCue struct {
```

Comment:
any other exciting options we should be offering based on what cue support?

relatedly, (we can do this in separate PR), we need to figure out how to inject values into cue (similar to --data-values-fie in ytt). how can we pass through PackageInstall's values to cue (values are represented as yaml there)? does cue project have an opinion on this?

Reply:
> any other exciting options we should be offering based on what cue support?

not that I know of, we could expose `--expression` to allow users to write cue expressions to transform their exported value, but that seemed like extra complexity without much justification right now.

>  how can we pass through PackageInstall's values to cue
I think we can have cue merge in yaml into the exported value, it works on the command-line if you explicitly specify the files like `cue export foo.yml foo.cue`, I don't see any way to automatically pickup files. The CLI is pretty confusing in its flags, it might make sense to just use the SDK to have more control.